### PR TITLE
Feat/redesign edit observation

### LINKED ISSUE #356: New / Edit Observation - No attachments
## New / Edit Observation - No attachments
Introducing the  new *Audio* feature by adding the **Microphone** icon as part of additional options to the **New / Edit Observation** screen. Therefore, this screen requires updates to be better display the additional options.

## Todo

Screen Shot 2024-05-15 at 1 55 23 PM

##  Update screen 
Update the following as designed :
Screen Shot 2024-05-15 at 2 34 58 PM

- [x] Update designs of **Title + Change**
- [x] Move and update the Coordinates, add the orange icon next to coordinates
- [x] Update the "what is happening here?" area
- [x] Update the bottom menu as designed
- [x] Update all the icons with a + 
- [x] Display Microphone icon as part of additional options

##  What is happening here? flow
- [x] When clicking on "what is happening here?", pull up the device keyboard
- [x] If keyboard open, if user taps anywhere else, bring back to the starting state

## Design
https://www.figma.com/design/1Knqv1NrijUuqMo1pvYyF7/Tracks-and-Audio-(Share)?node-id=529-916&t=V0vmc8SZM3TxMGuM-0

### PR DESCRIPTION
**Pull request by bohdanprog**
Closes #356

| Before | After |
|--------|--------|
|  |   |

### CODE REVIEW

File: src/frontend/screens/ObservationEdit/index.tsx [hunk 1]
```diff
@@ -57,25 +64,31 @@ export const ObservationEdit: NativeNavigationComponent<'ObservationEdit'> & {
...
     navigation.navigate('ObservationFields', {question: 1});
   }, [navigation]);
 
   const bottomSheetItems = [
     {
-      icon: <></>,
+      icon: <Audio width={30} height={30} />,
+      label: t(m.audioButton),
+      onPress: () => {},
+    },
```
Comment (ErikSin) [inline src/frontend/screens/ObservationEdit/index.tsx:80]: we should hide audio under a feature flag

File: src/frontend/sharedComponents/ActionTab/ActionTab.tsx [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+  icon: React.ReactNode;
+  label: string;
+  onPress: () => void;
+}
+
+export interface ActionProps {
+  items: ActionTabItems[];
+}
+
+export default function ActionTab({items}: ActionProps) {
```
Comment (ErikSin) [inline src/frontend/sharedComponents/ActionTab/ActionTab.tsx:17]: convert to named export (not default)

File: src/frontend/sharedComponents/ActionTab/ActionTab.tsx
Comment (ErikSin) [inline src/frontend/sharedComponents/ActionTab/ActionTab.tsx:1]: can you rename this file to `index.tsx` to follow conventions of the rest of the repo

File: src/frontend/sharedComponents/ActionTab/ItemButton.tsx [hunk 3]
```diff
@@ -0,0 +1,31 @@
+import React, {FC} from 'react';
+import {ActionTabItems} from './ActionTab';
+import {Pressable, StyleSheet, Text, View} from 'react-native';
+import {CustomCircleIcon} from './CustomCircleIcon';
+
+export const ItemButton: FC<ActionTabItems> = ({onPress, icon, label}) => (
+  <Pressable onPress={onPress} style={styles.itemContainer}>
```
Comment (ErikSin) [inline src/frontend/sharedComponents/ActionTab/ItemButton.tsx:7]: Can you change to the a `TouchableOpacity` please?

File: src/frontend/sharedComponents/ActionTab/KeyboardAccessory.tsx [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+
+interface KeyboardAccessory {
+  items: ActionTabItems[];
+  onPress: () => unknown;
+}
+export const KeyboardAccessory: FC<KeyboardAccessory> = ({onPress, items}) => (
+  <View style={[styles.flexRow, styles.containerPadding]}>
+    <View style={[styles.flexRow, styles.accessoryMainWrapper]}>
+      <Plus style={{marginRight: 10}} />
+      <Text style={styles.text}>Show Options</Text>
```
Comment (ErikSin) [inline src/frontend/sharedComponents/ActionTab/KeyboardAccessory.tsx:15]: this text is seeping into the icons on my device. TO fix this add `flex:1` to `styles.flexRow` and then in this text add `numberOfLines={1}`

Comment (ErikSin) [inline src/frontend/sharedComponents/ActionTab/KeyboardAccessory.tsx:15]: Also show options should be translated

File: src/frontend/screens/ObservationEdit/DescriptionField.tsx [hunk 5]
```diff
@@ -15,36 +16,75 @@ const m = defineMessages({
...
+
+    const keyboardShowUnsub = Keyboard.addListener('keyboardDidShow', () =>
+      setKeyboardVisible(true),
+    );
+
+    return () => {
+      keyboardHideUnsub.remove();
+      keyboardShowUnsub.remove();
+    };
+  }, []);
```
Comment (ErikSin) [inline src/frontend/screens/ObservationEdit/DescriptionField.tsx:36]: This is being used in 3 places with the exact same implementation . Can you make this a custom hook?

**Review by ErikSin — CHANGES_REQUESTED** [review]
Looks good, mostly aesthetic changes requested.

2 main things:
1. turn the keyboard listener into a custom hook
2. `images/redesign` can we rename the folder to `images/observationEdit`

File: src/frontend/screens/ObservationEdit/LocationView.tsx [hunk 6]
```diff
@@ -38,18 +38,13 @@ export const LocationView = () => {
 
   return (
     <View style={styles.locationContainer}>
-      {coordinateInfo.lat === undefined || coordinateInfo.lon === undefined ? (
+      {!coordinateInfo.lat || !coordinateInfo.lon ? (
```
[the remark this answers is not in the document]
Comment (bohdanprog) [reply src/frontend/screens/ObservationEdit/LocationView.tsx:41]: Right :)

File: src/frontend/screens/ObservationEdit/PresetInformation.tsx
[the remark this answers is not in the document]
Comment (bohdanprog) [reply src/frontend/screens/ObservationEdit/PresetInformation.tsx:1]: done

File: src/frontend/sharedComponents/ActionTab/ActionTab.tsx (hunk 2 above)
Comment (bohdanprog) [reply src/frontend/sharedComponents/ActionTab/ActionTab.tsx:17]: done

File: src/frontend/sharedComponents/ActionTab/ActionTab.tsx
Comment (bohdanprog) [reply src/frontend/sharedComponents/ActionTab/ActionTab.tsx:1]: good point, thanks

File: src/frontend/sharedComponents/ActionTab/ItemButton.tsx (hunk 3 above)
Comment (bohdanprog) [reply src/frontend/sharedComponents/ActionTab/ItemButton.tsx:7]: okay, no problem :)

File: src/frontend/sharedComponents/ActionTab/KeyboardAccessory.tsx (hunk 4 above)
Comment (bohdanprog) [reply src/frontend/sharedComponents/ActionTab/KeyboardAccessory.tsx:15]: done

Comment (bohdanprog) [reply src/frontend/sharedComponents/ActionTab/KeyboardAccessory.tsx:15]: Thanks for nit about that.
I fixed that, and now that should work properly.

File: src/frontend/screens/ObservationEdit/index.tsx (hunk 1 above)
Comment (bohdanprog) [reply src/frontend/screens/ObservationEdit/index.tsx:80]: Okay 👍  I called that: `EXPO_PUBLIC_FEATURE_AUDIO`
@ErikSin maybe we have to create some `.env.example`?

File: src/frontend/screens/ObservationEdit/DescriptionField.tsx (hunk 5 above)
Comment (bohdanprog) [reply src/frontend/screens/ObservationEdit/DescriptionField.tsx]: done

File: src/frontend/screens/Observation/ObservationHeaderRight.tsx
Comment (ErikSin) [inline src/frontend/screens/Observation/ObservationHeaderRight.tsx:1]: Just make sure that this is overwritten with what is on #365 (Aka, should default to false) before you merge this.

File: src/frontend/sharedComponents/ActionTab/KeyboardAccessory.tsx [hunk 7]
```diff
@@ -0,0 +1,61 @@
...
+
+const styles = StyleSheet.create({
+  flexRow: {
+    flexDirection: 'row',
+    alignItems: 'center',
+  },
+  text: {
+    fontSize: 20,
+    fontFamily: 'Rubik',
+    color: NEW_DARK_GREY,
```
Comment (ErikSin) [inline src/frontend/sharedComponents/ActionTab/KeyboardAccessory.tsx:49]: text also needs `flex:1` for the text shortening to work

**Review by ErikSin — APPROVED** [review]
One small fix (adding flex:1 to text).

And just make sure that #365 gets merged first, and that when you update from main the observation header is properly overwriten (aka it defaults to false)

File: src/frontend/sharedComponents/ActionTab/KeyboardAccessory.tsx (hunk 7 above)
Comment (bohdanprog) [reply src/frontend/sharedComponents/ActionTab/KeyboardAccessory.tsx:49]: done

**Pull request merged**